### ISSUE TITLE
georeferencing test fails with mapper 0.8.4 and libproj 5.2.0

### ISSUE THREAD

**Issue by slazav** [OpenOrienteering/mapper#1207]
Building of mapper 0.8.4 with libproj 5.2.0 fails at georeferencing test:
With my previous version of libproj (4.8.0) everything works fine.

$ ./georeferencing_t
********* Start testing of GeoreferencingTest *********
Config: Using QtTest library 5.11.3, Qt 5.11.3 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 8.2.1 20181011 (ALT Sisyphus 8.2.1-alt1))
PASS   : GeoreferencingTest::initTestCase()
PASS   : GeoreferencingTest::testEmptyProjectedCRS()
PASS   : GeoreferencingTest::testGridScaleFactor()
PASS   : GeoreferencingTest::testCRS(EPSG:4326)
PASS   : GeoreferencingTest::testCRS(UTM)
PASS   : GeoreferencingTest::testCRSTemplates()
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Koblenz UTM)
FAIL!  : GeoreferencingTest::testProjection(LVermGeo RLP Koblenz GK3) Compared values are not the same
   Actual   (QString::number(proj_coord.x(), 'f')): "3398096.685891"
   Expected (QString::number(easting, 'f'))       : "3398159.000000"
   Loc: [/home/sla/RPM/BUILD/openorienteering-mapper-0.8.3/test/georeferencing_t.cpp(209)]
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Pruem UTM)
FAIL!  : GeoreferencingTest::testProjection(LVermGeo RLP Pruem GK2) Compared values are not the same
   Actual   (QString::number(proj_coord.x(), 'f')): "2530524.326247"
   Expected (QString::number(easting, 'f'))       : "2530573.000000"
   Loc: [/home/sla/RPM/BUILD/openorienteering-mapper-0.8.3/test/georeferencing_t.cpp(209)]
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Landau UTM)
FAIL!  : GeoreferencingTest::testProjection(LVermGeo RLP Landau GK3) Compared values are not the same
   Actual   (QString::number(proj_coord.x(), 'f')): "3436686.345073"
   Expected (QString::number(easting, 'f'))       : "3436755.000000"
   Loc: [/home/sla/RPM/BUILD/openorienteering-mapper-0.8.3/test/georeferencing_t.cpp(209)]
PASS   : GeoreferencingTest::testProjection(EPSG 5514 д▄ц ZK Dolnц╜ Temenice)
PASS   : GeoreferencingTest::cleanupTestCase()
Totals: 11 passed, 3 failed, 0 skipped, 0 blacklisted, 5ms
********* Finished testing of GeoreferencingTest *********

**Comment by sikmir** [comment]
It works on ArchLinux, mapper 0.8.4 with proj 5.2.0.
```bash
$ pacman -Q | grep proj
proj 5.2.0-1
$ ./georeferencing_t 
********* Start testing of GeoreferencingTest *********
Config: Using QtTest library 5.12.1, Qt 5.12.1 (x86_64-little_endian-lp64 shared (dynamic) release build; by GCC 8.2.1 20181127)
PASS   : GeoreferencingTest::initTestCase()
PASS   : GeoreferencingTest::testEmptyProjectedCRS()
PASS   : GeoreferencingTest::testGridScaleFactor()
PASS   : GeoreferencingTest::testCRS(EPSG:4326)
PASS   : GeoreferencingTest::testCRS(UTM)
PASS   : GeoreferencingTest::testCRSTemplates()
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Koblenz UTM)
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Koblenz GK3)
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Pruem UTM)
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Pruem GK2)
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Landau UTM)
PASS   : GeoreferencingTest::testProjection(LVermGeo RLP Landau GK3)
PASS   : GeoreferencingTest::testProjection(EPSG 5514 ČÚZK Dolní Temenice)
PASS   : GeoreferencingTest::cleanupTestCase()
Totals: 14 passed, 0 failed, 0 skipped, 0 blacklisted, 5ms
********* Finished testing of GeoreferencingTest *********
```

**Comment by slazav** [comment]
Thanks! Looks like it is my local problem. I will try to investigate it.

**Comment by slazav** [comment]
Yes, it is not a problem of mapper (this issue can be closed), something wrong with my build of libproj.
I tried to reproduce the failed test and it looks like that no datum conversion happens with my proj 5.2.0:

gk3="+proj=tmerc +lat_0=0 +lon_0=9 +k=1.000000 +x_0=3500000 +y_0=0 +ellps=bessel +datum=potsdam +units=m +no-defs"
geo="+proj=latlong +datum=WGS84"
echo "7.567778 50.358944" | proj $gk3
echo "7.567778 50.358944" | cs2cs  $geo +to $gk3

with proj  4.8.0 gives me
3398096.70      5581182.12
3398158.37      5581313.53 -48.65

and with proj 5.2.0
3398096.70      5581182.12
3398096.70      5581182.12 0.00

**Comment by slazav** [comment]
... and this happens because in proj 5.2.0 a separate data file BETA2007.gsb should be packed in /usr/share/proj for this conversion. And I did not do this.
Thanks for your good tests!

**Comment by dg0yt** [comment]
Thanks for reporting! This is also [the build error I see for Fedora Rawhide](https://build.opensuse.org/package/live_build_log/home:dg0yt/openorienteering-mapper/Fedora_Rawhide/x86_64), but didn't have the time to investigate.

It is a little bit difficult to track, but BETA2007.gsb was first added to PROJ (osgeo/proj.4@3b95052) and then moved to proj-datumgrids (osgeo/proj.4@09f9727) before PROJ 5.0.0. So thanks to your report I learned that I will have to package proj-datumgrids when updating PROJ for Windows/macOS/Android. And for Fedora Rawhide, it is probably just a dependency.

There is now also a PROJ 6.0.0 RC. I contributed some last-minute changes to the CMake-controlled build, based on my recent attempt to package PROJ 5.2.0.
